### ISSUE TITLE
Train My Own Dataset

### ISSUE THREAD

**Issue by yifeipet** [beresandras/gan-flavours-keras#2]
Hello Andras,
I am very sorry to bother you again in your busy time. As you might know, many people might want to use their own dataset. But I used my COCO dataset (I split original images into 256x256 patches) and it gave me a Tensorflow error in the Augmenter. The error is not explicit. 

Since my dataset is very large, I use Keras API read image batch from directory. My dataset codes are:

df_train= tf.keras.preprocessing.image_dataset_from_directory(

    'Coco/train_patches',

    label_mode=None,

    color_mode="rgb",

    batch_size=32,

    image_size=(64,64),

    shuffle=True,

    seed=None,

)   #Found 235679 files belonging to 1 classes.

df_val= tf.keras.preprocessing.image_dataset_from_directory(

    'Coco/validation_patches',

    label_mode=None,

    color_mode="rgb",

    batch_size=32,

    image_size=(64,64),

    shuffle=True,

    seed=None,

) #Found 9860 files belonging to 1 classes.

train_ds = df_train.map(lambda x: x/255.0, num_parallel_calls = 1000).prefetch(200)
val_ds = df_val.map(lambda x: x/255.0, num_parallel_calls = 1000).prefetch(200)

I used the above codes in another Keras official example (https://keras.io/examples/generative/wgan_gp/) and there is no error. 
But I get the error when using my own dataset with your professional codes. 

I use Google Colab. 

Can you help me out? 

Thank you!

Sincerely,
Yifei

**Comment by beresandras** [comment]
What is the error message and the stack trace?

**Comment by yifeipet** [comment]
Hi Andras,
I will share the Colab link with you. Please wait. I need to check the codes again. Thank you!

Sincerely,
Yifei

**Comment by yifeipet** [comment]
Hi Andras,
I share the Colab file link with you: [https://colab.research.google.com/drive/1MR0cwd02LdRU_KIh_MaX9Zf37k-AHGjT?usp=sharing](url).
In the Colab codes, I include a link of the dataset. The dataset file size is 283 MB.
You can see the error that Tensorflow throw me in the Colab file. 
Please check the issue and help me out. Thank you!

For the link, just copy it and open it in a new tab. 

Sincerely,
Yifei

**Comment by beresandras** [comment]
Unfortunately, I do not have the time to debug it myself.

Based on the error, my guess is that the last batch is not the same size as the others, which causes an error. Unfortunately the image_dataset_from_directory() does not have an argument to drop the last batch.

train: 1683 // 128 = 13 complete batches
val: 748 // 128 = 5 complete batches

Put df_train.take(13) and df_val.take(5) into your code, before the map() calls.

**Comment by yifeipet** [comment]
Thank you, Andras! I should be able solve the dataset issue using a Tensorflow method not a Keras method.

**Comment by yifeipet** [comment]
Thank you, Andras! "Put df_train.take(13) and df_val.take(5) into your code, before the map() calls" solved the issue. 

Sincerely,
Yifei

**Comment by beresandras** [comment]
Happy to hear that. Closing the issue.
